src/hooks/useAppState/useAppState.test.js

```diff
@@ -15,6 +15,35 @@ const WrapperComponent = forwardRef((props, ref) => {
 const getState = (instance) => JSON.parse(instance.container.children[0].innerHTML);
 const clearDB = async () => Promise.all((await keys()).map((key) => del(key)));
 
+const renderAndHydrateComponent = async () => {
+  const ref = React.createRef();
+  const instance = render(<WrapperComponent ref={ref} />);
+  await waitFor(() => {
+    expect(instance.getByText(/Shopping List/)).toBeInTheDocument();
+  });
+
+  const state = {
+    index: 0,
+    items: [
+      [
+        { id: 's-1234', key: 's-1234', text: 'Apples' },
+        { id: 's-5678', key: 's-5678', text: 'Bananas' },
+      ],
+    ],
+    lists: [{ id: 'list-id', key: 'list-key', name: 'Shopping List' }],
+  };
+
+  await act(async () => {
+    await ref.current.actions.setState(state);
+  });
+
+  await waitFor(() => {
+    expect(getState(instance)).toEqual(state);
+  });
+
+  return { ref, instance };
+};
```

Comment:
This is just a helper function to render the component for our tests and add some items to the list